docs/ARCHITECTURE.md

```diff
@@ -52,38 +54,36 @@ output on to the next one.
 ```
 
 ### Shutdown
-To ensure that all tasks shutdown on completion (or failure) of the 
-simulation, we relay on the following: 
-1. [Triggered](https://docs.rs/triggered/latest/triggered): a `Trigger` 
-  that can be used to inform threads that it's time to shut down, and 
-  a `Listener` that propagates this signal.
-2. Channel mechanics: if all of the senders for a channel have been 
-  dropped (in our context, all of the producers have exited) then the 
-  corresponding receiving channel will error out on receiving. Likewise, 
-  if the receiving channel is dropped, the sending channels will error 
-  out.
-
-All events are handled in a `tokio::select` to allow waiting on 
-multiple asynchronous tasks at once. These selects should be `biased` 
-on the exit case (ie, the `Listener` being triggered) so that we 
+
+To ensure that all tasks shutdown on completion (or failure) of the
+simulation, we relay on the following:
+
+1. [Triggered](https://docs.rs/triggered/latest/triggered): a `Trigger`
+   that can be used to inform threads that it's time to shut down, and
+   a `Listener` that propagates this signal.
+2. The (`Trigger`, `Listener`) pair are used with channels: if a channel errors out across `send()` or `receive()`, shutdown is triggered. There is no reliance on channel mechanics, i.e. errors generated when all senders are and/or a receiver is dropped.
```

Comment:
I would clarify this to note that our channels don't have buffers, so we have to select on the shutdown signal on send/receive to make sure that a receiver exiting before a sender does doesn't indefinitely hang. 

Ie, this scenario: 

Task 1: Sending into `sender`
Task 2: Receiving on `receiver`

* Task 1 sends into `sender`, unblocks due to buffer size of 1 that we use everywhere
* Task 2 errors out before consuming from `receiver`
* Task 1 wants to send into `sender` again, but can't because the `receiver` has shut down 

If we always select on exit, then we don't run into this.